### PR TITLE
FIxed issue with wrong indices in the editor

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: editor/src/clj/editor/model_scene.clj [hunk 1]
```diff
@@ -110,17 +108,13 @@
...
 (defn mesh->vb! [^VertexBuffer vb ^Matrix4d world-transform vertex-space mesh scratch]
+  
   (let [^floats positions (to-scratch! mesh scratch :positions)
         ^floats normals   (to-scratch! mesh scratch :normals)
         ^floats texcoords (to-scratch! mesh scratch :texcoord0)
-        ^ints positions-indices (:position-indices mesh)
-        ^ints normals-indices   (:normals-indices mesh)
-        ^ints texcoords-indices (:texcoord0-indices mesh)
-        vcount (alength positions-indices)
+        ^ints indices (:indices mesh)
```
Comment (JCash) [inline editor/src/clj/editor/model_scene.clj:116]: The issue was that we have switched to a regular index buffer.

File: editor/src/clj/editor/model_scene.clj [hunk 2]
```diff
@@ -143,27 +137,23 @@
...
-            (.put fb texcoords (umul 2 ti) 2)))))
+      (dotimes [i vcount]
+        (let [vi (aget indices i)
+              normal (float-array [0.0 0.0 -1.0])]
+          (.put fb positions (umul 3 vi) 3)
+          (if (not= (alength normals) 0)
+            (.put fb normals (umul 3 vi) 3)
+            (.put fb normal 0 3))
+          (.put fb texcoords (umul 2 vi) 2)
+          )))
```
Comment (JCash) [inline editor/src/clj/editor/model_scene.clj:155]: All this fluff in the editor is cumbersome.
I think we should pack as much data from C++ given the desired vertex declaration.

Opening a small model (~4000 vertices) takes a significant amount of time.
Loading the model in C++ takes ~0.001s, so we need to measure in the java+clojure code next. Not a new problem afaict, so we'll add a separate ticket for it.

File: editor/src/clj/editor/model_scene.clj [hunk 3]
```diff
@@ -261,14 +251,31 @@
 (defn- doubles->floats [ds]
   (float-array (map float ds)))
 
+(defn- bytes-to-indices [^ByteString indices index-format]
+  (let [num-bytes (.size indices)
+        ba (byte-array num-bytes)
+        _ (.copyTo indices ba 0)
+        bb (ByteBuffer/wrap ba)
```
Comment (JCash) [inline editor/src/clj/editor/model_scene.clj:258]: DIdn't manage to get a readable byte byffer from the com.google.protobuf.ByteString, so this copy had to be done for now.

File: editor/src/clj/editor/model_scene.clj [hunk 4]
```diff
@@ -261,14 +251,31 @@
...
+             (.asShortBuffer bb)
+             (.asIntBuffer bb))
+
+        ;; Make it into a human readable/printable int array
+        count (if (= :indexbuffer-format-16 index-format)
+                (/ num-bytes 2)
+                (/ num-bytes 4))
+        ia (int-array count)
+        _ (dotimes [i count]
+            (aset ia i (int (.get bo i))))]
```
Comment (JCash) [inline editor/src/clj/editor/model_scene.clj:270]: It is just too annoying to printout values and getting weird java structs back.

File: engine/modelc/src/test/test_model.cpp [hunk 5]
```diff
@@ -95,9 +96,14 @@ TEST(ModelGLTF, LoadSkeleton)
...
 static int TestStandalone(const char* path)
 {
+    uint64_t tstart = dmTime::GetTime();
+
     dmModelImporter::Options options;
     dmModelImporter::Scene* scene = LoadScene(path, options);
 
+    uint64_t tend = dmTime::GetTime();
+    printf("Model %s loaded in %.3f seconds.\n", path, float(tend-tstart)/1000000.0f);
+
```
Comment (JCash) [inline engine/modelc/src/test/test_model.cpp:106]: Our model test doubles as a first line of testing the importing a single file and outputting the statistics.
Easy to debug and run, and always available.

File: editor/src/clj/editor/model_scene.clj (hunk 2 above)
Comment (matgis) [reply editor/src/clj/editor/model_scene.clj:155]: Yeah, we should be able to just prepare the buffers in the cpp code and use something like `NewDirectByteBuffer` to pass the data directly to the `.glBufferData` method. I'm not sure if we should be using the `VertexBuffer` record for this, but we definitely shouldn't be populating it like this. Perhaps we should just be creating a new `VertexBuffer` wrapping the `ByteBuffer` returned from cpp whenever there's an update. Not sure yet, but we shouldn't need to do much work at all on the editor side.

https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#NewDirectByteBuffer

**Pull request merged**